src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1221,7 +1223,16 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement=="invokedynamic")
     {
       // not used in Java
-      code_typet &code_type=to_code_type(arg0.type());
+      code_typet &code_type = to_code_type(arg0.type());
+
+      const java_class_typet::java_lambda_method_handlet &lambda_method_handle =
+        lambda_method_handles.at(
+          std::stoi(id2string(code_type.get(ID_java_lambda_method_handle_id))));
```

Comment:
`index` rather than `id` since I think that more strongly suggests it is a zero indexed counter rather than some unique identifier.